shark_turbine/kernel/wave/codegen.py

```diff
@@ -412,9 +459,48 @@ def handle_write(emitter: WaveEmitter, node: fx.Node):
 ###############################################################################
 
 
+def emit_mfma(
+    m: int, n: int, k: int, vector_type: VectorType, acc: Value, values: list[Value]
+):
+    m = get_constant_attr(m, IntegerType.get_signless(32))
+    n = get_constant_attr(n, IntegerType.get_signless(32))
+    k = get_constant_attr(k, IntegerType.get_signless(32))
+    blocks = get_constant_attr(1, IntegerType.get_signless(32))
+
+    result = amdgpu_d.mfma(
+        dest_d=vector_type,
+        m=m,
+        n=n,
+        k=k,
+        blocks=blocks,
+        source_a=values[0],
+        source_b=values[1],
+        dest_c=acc,
+    )
+    return result
+
+
 @handle_op(mma)
 def handle_mma(emitter: WaveEmitter, node: fx.Node):
-    raise NotImplementedError("MMA: Currently only stub implementation")
+    try:
+        lhs, rhs, acc = node.args
+        acc = cast_vector(emitter, acc)
+        values = [lhs, rhs]
```

Comment:
`values = [cast_vector(emitter, val) for val in [lhs, rhs]]`

Reply:
thanks